disnake/member.py

```diff
@@ -1004,6 +1004,45 @@ async def edit(
             data = await http.edit_member(guild_id, self.id, reason=reason, **payload)
             return Member(data=data, guild=self.guild, state=self._state)
 
+    async def fetch_voice_state(self, *, cache: bool = False) -> VoiceState:
+        """|coro|
+
+        Fetches the :class:`VoiceState` of the member.
+
+        .. versionadded:: 2.10
+
+        .. warning::
+
+            If the :attr:`disnake.Intents.voice_states` is disabled caching this could lead to a memory leak.
+            This could happen because with the intent disabled you don't receive voice state updates, hence
+            the cached voice state could possibly never be removed from the cache.
+
+        Parameters
+        ----------
+        cache: :class:`bool`
```

Comment:
After giving this some more thought, I think ultimately I'd prefer leaving caching up to users and just have this method simply return the voice state, since purging the cache correctly is essentially impossible here